projects/demo/src/app/breadcrumbs/breadcrumb.demo.service.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016-2025 Broadcom. All Rights Reserved.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+
+import { MenuItem } from './breadcrumbs.demo.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BreadcrumbDemoService {
+  breadcrumbs: BehaviorSubject<MenuItem[]> = new BehaviorSubject<MenuItem[]>([]);
+
+  createBreadcrumbs(route: ActivatedRoute, url = '', breadcrumbs: MenuItem[] = []): MenuItem[] {
```

Comment:
This should just a be `function` as discussed in the other thread. There is no need for a class here. And it doesn't really need to be a separate file either.